File: app/src/main/java/com/kabirnayeem99/paymentpaid/fragments/PaymentsFragment.java
package com.kabirnayeem99.paymentpaid.fragments;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.kabirnayeem99.paymentpaid.R;
import com.kabirnayeem99.paymentpaid.WorkViewModel;
import com.kabirnayeem99.paymentpaid.adapters.PaymentAdapter;
import com.kabirnayeem99.paymentpaid.utils.CustomUtils;

import java.util.List;


public class PaymentsFragment extends Fragment {
    private static final String TAG = "PaymentsFragment";
    RecyclerView rvPaymentListByMonth;
    TextView tvPaymentTotal;
    PaymentAdapter paymentAdapter;


    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_payments, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {

        initViews(view);
        WorkViewModel workViewModel = ViewModelProviders.of(requireActivity()).get(WorkViewModel.class);
        paymentAdapter = new PaymentAdapter();

        workViewModel.getTotalPaymentByMonth(CustomUtils.getCurrentYear()).observe(requireActivity(), new Observer<List<Integer>>() {
            @Override
            public void onChanged(List<Integer> integers) {
                paymentAdapter.setMonthlyPaymentList(integers);
            }
        });

        workViewModel.getTotalPaymentByYear().observe(requireActivity(),
                integer -> tvPaymentTotal.setText(String.valueOf(integer == null ? 0 : integer)));
        initRecyclerView();

        super.onViewCreated(view, savedInstanceState);
    }

    private void initViews(@NonNull View view) {
        rvPaymentListByMonth = view.findViewById(R.id.rv_payment_list_by_month_payments);
        tvPaymentTotal = view.findViewById(R.id.tv_payment_list_total_payments);
    }

    private void initRecyclerView() {

        rvPaymentListByMonth.setLayoutManager(new LinearLayoutManager(requireActivity()));
        rvPaymentListByMonth.setHasFixedSize(true);
        rvPaymentListByMonth.setAdapter(paymentAdapter);
    }

}

File: app/src/main/java/com/kabirnayeem99/paymentpaid/ui/fragments/PaymentsFragment.kt
package com.kabirnayeem99.paymentpaid.ui.fragments

import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import com.kabirnayeem99.paymentpaid.R
import com.kabirnayeem99.paymentpaid.adapters.PaymentAdapter
import com.kabirnayeem99.paymentpaid.ui.FirestoreViewModel
import com.kabirnayeem99.paymentpaid.ui.activities.HomeActivity
import kotlinx.android.synthetic.main.fragment_payments.*

class PaymentsFragment : Fragment(R.layout.fragment_payments) {

    companion object {
        const val TAG = "PaymentsFragment"
    }

    private lateinit var paymentAdapter: PaymentAdapter
    private lateinit var firestoreViewModel: FirestoreViewModel


    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {

        super.onViewCreated(view, savedInstanceState)

        firestoreViewModel = (activity as HomeActivity).firestoreViewModel

        initRecyclerView()

        manipulateData()

    }

    private fun manipulateData() {

        Log.d(tag, "manipulateData: the manipulating data started")
        firestoreViewModel.paymentList.observe(viewLifecycleOwner,
                { paymentList ->
                    Log.d(tag, "manipulateData: $paymentList")
                    paymentList?.let {
                        paymentAdapter.differ.submitList(paymentList)
                    }
                })

        firestoreViewModel.paymentOfCurrentYear.observe(viewLifecycleOwner,
                { totalPayment ->
                    totalPayment?.let {
                        tvTotalPayment.text = totalPayment.toString()
                    }
                })
    }


    private fun initRecyclerView() {

        paymentAdapter = PaymentAdapter()

        rvPaymentListByMonth.apply {
            adapter = paymentAdapter
            layoutManager = LinearLayoutManager(activity)
        }
    }

}